packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -151,11 +151,11 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
           overlayEntries.first.opaque = false;
         break;
       case AnimationStatus.dismissed:
-        // We might still be the current route if a subclass is controlling the
+        // We might still be an active route if a subclass is controlling the
         // the transition and hits the dismissed status. For example, the iOS
         // back gesture drives this animation to the dismissed status before
-        // popping the navigator.
-        if (!isCurrent) {
+        // removing the route.
```

Comment:
Add that it itself is responsible for disposing the route and we don't have to do it here.

Reply:
Done.